### PR TITLE
Updating Endpoints to meet Spec

### PR DESCRIPTION
**Pull request by sublet**
Following the [API Spec](https://pathcheck.atlassian.net/wiki/spaces/SA/pages/73924822/SPL+API+Specification+v1+For+MVP1), I present to you this fun PR.

Moving trails to points.
Removes Redacted API Endpoints
Removes SafePaths API Endpoints
Adds all `/organization/cases` endpoints.
Adds all `/case` endpoints.
Adds all `/point` endpoints.
Adds GCS integration.

Tests to go with it.
Models have been updated to handle requests.

### CODE REVIEW

File: db/migrations/20200526095904_rebuild.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ function buildOrganization(knex) {
 function buildSettings(knex) {
   return knex.schema.createTable('settings', function (table) {
     table.uuid('id').notNull().primary();
-    table.uuid('organization_id').notNull().references('organizations.id').onDelete('CASCADE');
+    table.integer('organization_id').notNull().references('organizations.id').onDelete('CASCADE');
```
Comment (MarshallMcCoy) [inline db/migrations/20200526095904_rebuild.js:27]: We need a field named `privacy_policy_url` to capture where the org's privacy policy is stored.

Comment (sublet) [reply db/migrations/20200526095904_rebuild.js:27]: Done

File: app/api/case/controller.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+ * Captures user consent to having their data published in the 
+ * aggregated anonymized JSON file that is available to public.
+ *
+ */
+exports.consentToPublish = async (req, res) => {
+  const { caseId } = req.body;
+
+  if (!caseId) throw new Error('Case ID is not valid.')
+
+  const data = {
```
Comment (MarshallMcCoy) [inline app/api/case/controller.js:62]: This endpoint doesn't seem to actually persist anything? I realize you might be planning taking care of that at a later date. If so, mind adding a `TODO`?

File: app/api/case/controller.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+  if (!caseIds) throw new Error('Case IDs are invalid.')
+  if (!organization_id) throw new Error('Organization ID is not valid.')
+
+  const organization = await organizationsService.fetchById(organization_id);
+  if (organization) {
+    const publishResults = await casesService.publishCases(caseIds, organization.id)
+
+    const points = await casesService.fetchAllPublishedPoints()
+    if (points && points.length > 0) {
```
Comment (MarshallMcCoy) [inline app/api/case/controller.js:125]: Seems like we're doing a lot in this endpoint. A little concerned running all these operations on a large data set could take an extended period of time and the client might end up timing out. Not anything that we need to handle for right now but curious to see what comes up in testing.

File: db/models/points.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    const points = await this.find({ case_id })
+    if (points) {
+      return this._getRedactedPoints(points)
+    }
+    throw new Error('Could not find redacted points.')
+  }
+
+  async createRedactdPoint(caseId, point) {
+    let record = {};
+    let hash = await geoHash.encrypt(point)
```
Comment (MarshallMcCoy) [inline db/models/points.js:31]: This (and the same function call in the update action below) takes care of [this ticket](https://pathcheck.atlassian.net/browse/PLACES-241), right?

File: db/migrations/20200526095904_rebuild.js [hunk 5]
```diff
@@ -14,7 +14,7 @@ function dropTables(knex) {
 
 function buildOrganization(knex) {
   return knex.schema.createTable('organizations', function (table) {
-    table.uuid('id').notNull().primary();
+    table.increments('id').notNull().primary();
```
Comment (MarshallMcCoy) [inline db/migrations/20200526095904_rebuild.js:17]: Can we also add a `completed_onboarding` flag (defaults to `false`) to the `organizations` table? If so, it'll knock out [this ticket](https://pathcheck.atlassian.net/browse/PLACES-244) 😉

**Comment by aiyan** [comment]
This is a monolithic PR 🥇 😊. LGTM.

File: app/api/case/controller.js (hunk 3 above)
Comment (sublet) [reply app/api/case/controller.js:125]: On point.  I could see something that Publishes everything, then a cron goes in and generates the Points and files.  It should be pretty fast, but depending on what the testing shows, we may need to refactor a bit.

File: db/models/points.js (hunk 4 above)
Comment (sublet) [reply db/models/points.js:31]: Correct.

File: app/api/case/controller.js (hunk 2 above)
Comment (sublet) [reply app/api/case/controller.js:62]: Planning to handle later once the API spec is finalized for it.  Is currently in a "Needs Confirmation" state

Comment (sublet) [reply app/api/case/controller.js:62]: Will add todo.

File: db/migrations/20200526095904_rebuild.js (hunk 5 above)
Comment (sublet) [reply db/migrations/20200526095904_rebuild.js:17]: Done

File: app/api/case/controller.js (hunk 2 above)
Comment (sublet) [reply app/api/case/controller.js:65]: Done

**Comment by sublet** [comment]
@sherifkozman Looks like tests are failing, and just need tp be updated for the new endpoints.

**Comment by sherifkozman** [comment]
@AdamLeonSmith Can you please confirm updating tests so we can merge this PR by @sublet

**Comment by AdamLeonSmith** [comment]
@alfonsoPQA can you remove Test 005 on this branch?

**Pull request merged**